File: app/src/main/java/com/petnbu/petnbu/notification/NotificationsFragment.java
package com.petnbu.petnbu.notification;

import android.arch.lifecycle.ViewModelProviders;
import android.databinding.DataBindingUtil;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.support.v7.widget.LinearLayoutManager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.petnbu.petnbu.R;
import com.petnbu.petnbu.databinding.FragmentNotificationsBinding;

import java.util.ArrayList;

public class NotificationsFragment extends Fragment {

    private NotificationViewModel mNotificationViewModel;

    private FragmentNotificationsBinding mBinding;
    private NotificationsRecyclerViewAdapter mAdapter;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
    }

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        mBinding = DataBindingUtil.inflate(inflater, R.layout.fragment_notifications, container, false);
        initialize();
        return mBinding.getRoot();
    }

    private void initialize() {
        mNotificationViewModel = ViewModelProviders.of(this).get(NotificationViewModel.class);
        mBinding.setViewModel(mNotificationViewModel);
        mNotificationViewModel.getNotifications().observe(this, notifications -> mAdapter.setData(notifications));

        mBinding.rvNotifications.setLayoutManager(new LinearLayoutManager(getActivity()));
        mAdapter = new NotificationsRecyclerViewAdapter(getActivity(), new ArrayList<>());
        mBinding.rvNotifications.setAdapter(mAdapter);
    }
}

File: app/src/main/java/com/petnbu/petnbu/notification/NotificationsFragment.kt
package com.petnbu.petnbu.notification

import android.arch.lifecycle.Observer
import android.arch.lifecycle.ViewModelProviders
import android.content.Context
import android.databinding.DataBindingUtil
import android.os.Bundle
import android.support.v4.app.Fragment
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

import com.petnbu.petnbu.R
import com.petnbu.petnbu.databinding.FragmentCommentsBinding
import com.petnbu.petnbu.databinding.FragmentNotificationsBinding
import com.petnbu.petnbu.model.NotificationUI
import kotlinx.android.synthetic.main.fragment_notifications.*

import java.util.ArrayList

class NotificationsFragment : Fragment() {

    private lateinit var mNotificationViewModel: NotificationViewModel
    private lateinit var mBinding: FragmentNotificationsBinding

    private var mAdapter: NotificationsRecyclerViewAdapter? = null

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        mBinding = DataBindingUtil.inflate(inflater, R.layout.fragment_notifications, container, false)
        initialize()
        return mBinding.root
    }

    private fun initialize() {
        activity?.let {
            mNotificationViewModel = ViewModelProviders.of(it).get(NotificationViewModel::class.java)
            mBinding.viewModel = mNotificationViewModel
            mNotificationViewModel.loadNotifications().observe(this, Observer { notifications ->
                if (notifications != null) {
                    mAdapter?.submitList(notifications)
                }
            })

            mBinding.rvNotifications.layoutManager = LinearLayoutManager(activity)
            mAdapter = NotificationsRecyclerViewAdapter(context = it)
            mBinding.rvNotifications.adapter = mAdapter
        }

    }
}
